### PR TITLE
[PyTorch] Add word2vec pretraining

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1498/4 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1498/

**Comment by AnirudhDagar** [comment]
@astonzhang I also changed one line in MXNet with this PR.
```
loss = gluon.loss.SigmoidBinaryCrossEntropyLoss() -> loss = gluon.loss.SigmoidBCELoss()	
```
Just a simpler alias imo and also more similar to what it's called in other frameworks (eg PyTorch: `BCEWithLogitsLoss`). Let me know

File: chapter_natural-language-processing-pretraining/word2vec-pretraining.md [hunk 1]
```diff
@@ -26,47 +37,77 @@ We will implement the skip-gram model by using embedding layers and minibatch mu
...
+```{.python .input}
 embed = nn.Embedding(input_dim=20, output_dim=4)
 embed.initialize()
 embed.weight
 ```
 
+```{.python .input}
+#@tab pytorch
+embed = nn.Embedding(num_embeddings=20, embedding_dim=4)
+print(f"Parameter embedding_weight ({embed.weight.shape}, "
```
Comment (astonzhang) [inline chapter_natural-language-processing-pretraining/word2vec-pretraining.md:49]: Let's consistently use single quote in f-string

**Comment by astonzhang** [comment]
> @astonzhang I also changed one line in MXNet with this PR.
> 
> ```
> loss = gluon.loss.SigmoidBinaryCrossEntropyLoss() -> loss = gluon.loss.SigmoidBCELoss()	
> ```
> 
> Just a simpler alias imo and also more similar to what it's called in other frameworks (eg PyTorch: `BCEWithLogitsLoss`). Let me know if you want me to change it back.

Using simpler alias looks good. Thanks!

File: chapter_natural-language-processing-pretraining/word2vec-pretraining.md (hunk 1 above)
Comment (astonzhang) [reply chapter_natural-language-processing-pretraining/word2vec-pretraining.md:49]: i'll fix this

Comment (AnirudhDagar) [reply chapter_natural-language-processing-pretraining/word2vec-pretraining.md:49]: Sure! I missed that, thanks for the fix :)
